### PR TITLE
ndb.key.urlsafe() should return a string not bytes #71

### LINKED ISSUE #71: ndb.key.urlsafe() should return a string not bytes
## Expected Behavior
When calling ndb.key.urlsafe() you should get a string that can be concatenated with other strings. 

## Actual Behavior
Currently it return bytes so you need to add .decode() on every call to urlsafe()

## Steps to Reproduce the Problem

1. On a ndb.Model derived class do:  x.key.urlsafe() + '\'
2. this will generate a runtime error: TypeError: can only concatenate str (not "bytes") to str

## Specifications

- Version: 1.0.0
- Platform: Ubuntu 20.04 (wsl2)

### PR DESCRIPTION
**Pull request by martin-green**
Fixes #71 

- [x] Tests pass

### CODE REVIEW

**Comment by ludoch** [comment]
Hi,
Thanks for the PR. Can you update the missing CLA step?

**Comment by martin-green** [comment]
Done

Den 2023-03-28 kl. 20:37, skrev Ludovic Champenois:

**Comment by kamilturek** [comment]
Hey @ludoch, is there any update on this? I am curious about this change as it would mean drifting away from the Cloud NDB behaviour (which also returns bytes).

https://github.com/googleapis/python-ndb/blob/main/MIGRATION_NOTES.md#:~:text=Key.urlsafe%20outputs,in%20NDB%20behavior.

**Comment by shreejad** [comment]
Please refer to https://github.com/GoogleCloudPlatform/appengine-python-standard/issues/71#issuecomment-1599355090

TL;DR: This is a breaking change and will create divergent behaviors with Cloud NDB, who also choose to return bytes.
